Brighter images

Chr1st0p43rR opened the pull request:
A couple of changes:

1. I added an option, `--adjust_image`, for which you can specify `auto` or `gamma` (default is None), which specifies adjustment of the representative slice of the input image accordingly. Note:
a) I have stuck to limiting adjustment to the input image, because I think it is safe to assume that this image will not be binary, whereas this assumption is not so safe for the reference annotations and prediction images. Of course, further processing of these images might be what we want to do in the future, but am thinking that currently, it is beyond the scope of this PR, given our deadline.
b) the two options, auto and gamma, use Pillow's ImageOps.autocontrast and a home-made gamma correction, respectively. To not overcomplicate things at this stage, the gamma correction has a hardcoded value.
2. I fixed the file paths for images in the model_dataset_table json.

Chr1st0p43rR committed:
Added optional output image histogram adjustment.

Chr1st0p43rR committed:
Updated file paths for example images.

Chr1st0p43rR committed:
Added explicit conversion to 8-bit before brightness adjustment.

Chr1st0p43rR committed:
Added text file with image-model combinations for testing, and referenced that in local batch script.

tzulueta commented:
Hi Chris, everything looks fine and the approach seems sensible. Just a few comments:

- I agree with your comment on this PR to limit the brightness adjustment to the input image. However if `adjust_image` is given a value, the prediction is also adjusted, and the adjusted prediction is saved. This makes the prediction look very different in cases when we do image restoration (like denoising). I'd probably avoid adjusting the prediction in all cases. 

- Should we update the README explaining the two options for image brightness adjustment (gamma and auto) a little bit better? At least we should give some more info in `run_benchmark_bmz_on_bia_data.py`:
`@click.option("-adj_img", "--adjust_image", type=str, default=None, 
              help="optional brightness adjustment of the output image; can choose auto or gamma")`
So it's obvious that only the input image will be adjusted if somebody checks the --help command. 

- Should we include the instructions to run the repo using poetry in the README?

- We need to pin the xarray version (to 2025.1.2) in the conda and conda-dev env.yaml file, and in the poetry pyproject.toml file. 

Let me know if something needs more discussion!

Chr1st0p43rR committed:
Removed adjustment of prediction image.

Chr1st0p43rR committed:
Added documentation for image adjustment in --help and in readme.

Chr1st0p43rR committed:
Added pinned xarray in dev env.

Chr1st0p43rR commented:
Hi Teresa,

Thank you for the suggestions. Following your ordering:

- I've removed the adjustment of the prediction image.
- I have updated the readme and the --help to explain how to use the adjustment option.
- There is a separate branch (here: https://github.com/BioImage-Archive/bia-bmz-integration/tree/environment_fix) for using poetry — I had been working on that, but now I suppose we need to discuss what we want to do with it, since we are having fewer problems with conda.
- The version of xarray in the main conda env yaml is already pinned; think that was in one of your PRs. I have added the version spec. to the dev env yaml. I think it is not necessary in the pyproject file, since that follows the installations in the env file — I have checked by updating my dev environment with the pinned xarray, and I get the "requirement already satisfied" message for xarray (with the correct version!).

Let me know what you think; happy to discuss / make more changes. Thanks!

tzulueta commented:
Thanks for the changes, everything looks good to me!
